### PR TITLE
Feature/arc 1017

### PR DESCRIPTION
**Pull request by ian-emsens-shd**
https://meemoo.atlassian.net/browse/ARC-1017

### CODE REVIEW

File: src/modules/shared/components/TagSearchBar/TagSearchBar.tsx [hunk 1]
```diff
@@ -26,23 +26,26 @@ const components = {
...
+	options,
+	searchValue,
+	size = undefined,
+	syncSearchValue = true,
+	valuePlaceholder,
 	...tagsInputProps
 }: TagSearchBarProps<IsMulti>): ReactElement => {
-	const [inputValue, setInputValue] = useState(searchValue);
+	const localInputState = useState(searchValue);
+	const [inputValue, setInputValue] = inputState || localInputState;
```
Comment (ian-emsens-shd) [inline src/modules/shared/components/TagSearchBar/TagSearchBar.tsx:47]: Can't avoid double state here due to the way React demands hooks to work

File: src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx [hunk 2]
```diff
@@ -576,17 +590,18 @@ const VisitorSpaceSearchPage: NextPage = () => {
 									clearLabel={t(
 										'pages/bezoekersruimte/slug___wis-volledige-zoekopdracht'
 									)}
+									inputState={searchBarInputState}
```
Comment (bertyhell) [inline src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx:593]: optional: might be cleaner to destructure the useState and reconstruct it here
so the other usages of searchBarInputState are more clear:

```
const [searchBarInput, setSearchBarInput] = useState<string>();

const searchValue = prepareSearchValue(searchBarInput);

setSearchBarInput(undefined);

inputState={[searchBarInput, setSearchBarInput]}
```

File: src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx [hunk 3]
```diff
@@ -235,13 +236,24 @@ const VisitorSpaceSearchPage: NextPage = () => {
 	 * Methods
 	 */
 
+	const prepareSearchValue = (value = '') => {
+		if (value.trim() && !query.search?.includes(value)) {
```
Comment (bertyhell) [inline src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx:240]: also trim any whitespace before searching in query.search
```suggestion
		if (value.trim() && !query.search?.includes(value.trim())) {
```

File: src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx [hunk 4]
```diff
@@ -235,13 +236,24 @@ const VisitorSpaceSearchPage: NextPage = () => {
 	 * Methods
 	 */
 
+	const prepareSearchValue = (value = '') => {
```
Comment (bertyhell) [inline src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx:239]:
```suggestion
	const prepareSearchValue = (value = ''): Record<string, string> | undefined => {
```

File: src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx [hunk 5]
```diff
@@ -235,13 +236,24 @@ const VisitorSpaceSearchPage: NextPage = () => {
...
+		if (value.trim() && !query.search?.includes(value)) {
+			return {
+				[SEARCH_QUERY_KEY]: (query.search ?? []).concat(value),
+			};
+		}
+
+		return undefined;
```
Comment (bertyhell) [inline src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx:246]: personal preference: 
When i set something explicitly i prefer to use null.
when something is missing without setting it explicitly it is undefined.

**Review by bertyhell — APPROVED** [review]

File: src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx (hunk 5 above)
Comment (ian-emsens-shd) [reply src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx:247]: best to illustrate why I prefer `undefined` over `null` across the board.
image

Comment (bertyhell) [reply src/modules/visitor-space/components/VisitorSpaceSearchPage/VisitorSpaceSearchPage.tsx:251]:
```
let nullable: { prop?: string | null }
```

**Pull request merged**